diff -ur xf86-video-intel-2.99.917/src/sna/sna_accel.c xf86-video-intel-2.99.917-pc/src/sna/sna_accel.c
--- xf86-video-intel-2.99.917/src/sna/sna_accel.c	2021-11-03 16:24:27.669484747 +0100
+++ xf86-video-intel-2.99.917-pc/src/sna/sna_accel.c	2021-11-03 16:22:30.753591499 +0100
@@ -17672,7 +17672,7 @@
 			continue;
 
 		src = dirty->src;
-		dst = dirty->slave_dst->master_pixmap;
+		dst = dirty->secondary_dst->primary_pixmap;
 
 		region.extents.x1 = dirty->x;
 		region.extents.x2 = dirty->x + dst->drawable.width;
@@ -17699,7 +17699,7 @@
 		dy += dirty->dst_y;
 #endif
 		RegionTranslate(&region, dx, dy);
-		DamageRegionAppend(&dirty->slave_dst->drawable, &region);
+		DamageRegionAppend(&dirty->secondary_dst->drawable, &region);
 
 		DBG(("%s: slave:  ((%d, %d), (%d, %d))x%d\n", __FUNCTION__,
 		     region.extents.x1, region.extents.y1,
@@ -17776,7 +17776,7 @@
 			kgem_bo_sync__gtt(&sna->kgem, __sna_pixmap_get_bo(dst));
 		}
 
-		DamageRegionProcessPending(&dirty->slave_dst->drawable);
+		DamageRegionProcessPending(&dirty->secondary_dst->drawable);
 skip:
 		RegionUninit(&region);
 		DamageEmpty(dirty->damage);
diff -ur xf86-video-intel-2.99.917/src/uxa/intel_driver.c xf86-video-intel-2.99.917-pc/src/uxa/intel_driver.c
--- xf86-video-intel-2.99.917/src/uxa/intel_driver.c	2021-11-03 16:24:27.676151157 +0100
+++ xf86-video-intel-2.99.917-pc/src/uxa/intel_driver.c	2021-11-03 16:23:43.264480387 +0100
@@ -624,30 +624,30 @@
 	RegionRec pixregion;
 	int was_blocked;
 
-	PixmapRegionInit(&pixregion, dirty->slave_dst->master_pixmap);
+	PixmapRegionInit(&pixregion, dirty->secondary_dst->primary_pixmap);
 	RegionTranslate(&pixregion, dirty->x, dirty->y);
 	RegionIntersect(&pixregion, &pixregion, DamageRegion(dirty->damage));
 	RegionTranslate(&pixregion, -dirty->x, -dirty->y);
 	was_blocked = RegionNil(&pixregion);
-	DamageRegionAppend(&dirty->slave_dst->drawable, &pixregion);
+	DamageRegionAppend(&dirty->secondary_dst->drawable, &pixregion);
 	RegionUninit(&pixregion);
 	if (was_blocked)
 		return;
 
-	PixmapRegionInit(&pixregion, dirty->slave_dst->master_pixmap);
+	PixmapRegionInit(&pixregion, dirty->secondary_dst->primary_pixmap);
 	PixmapSyncDirtyHelper(dirty, &pixregion);
 	RegionUninit(&pixregion);
 
         intel_flush(intel);
 	if (!intel->has_prime_vmap_flush) {
-		drm_intel_bo *bo = intel_uxa_get_pixmap_bo(dirty->slave_dst->master_pixmap);
+		drm_intel_bo *bo = intel_uxa_get_pixmap_bo(dirty->secondary_dst->primary_pixmap);
 		was_blocked = xf86BlockSIGIO();
 		drm_intel_bo_map(bo, FALSE);
 		drm_intel_bo_unmap(bo);
 		xf86UnblockSIGIO(was_blocked);
 	}
 
-	DamageRegionProcessPending(&dirty->slave_dst->drawable);
+	DamageRegionProcessPending(&dirty->secondary_dst->drawable);
 	return;
 }
 
